hw/ip/aes/pre_syn/syn_setup.example.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Setup IP name and top module.
+# When changing the top module, some parameters might not exist and
+# tcl/yosys_run_synth.tcl might need to be adjusted accordingly.
+export LR_SYNTH_IP_NAME=aes
+export LR_SYNTH_TOP_MODULE=aes
+
+# Setup module parameters.
+export LR_SYNTH_AES_192_ENABLE=1
+export LR_SYNTH_MASKING=1
+export LR_SYNTH_S_BOX_IMPL=4
+
+# Setup cell library path.
+# Uncomment the lines below and set the path to an appropriate .lib file
+# export LR_SYNTH_CELL_LIBRARY_PATH=/path/to/NangateOpenCellLibrary_typical.lib
+# export LR_SYNTH_CELL_LIBRARY_NAME=nangate
+
+# Control synthesis flow.
+export LR_SYNTH_TIMING_RUN=0
+export LR_SYNTH_FLATTEN=1
+
+# For timing runs, hierarchy flattening is needed.
+if [[ $LR_SYNTH_TIMING_RUN == 1 ]] && [[ $LR_SYNTH_FLATTEN == 0 ]] ;
+then
+  echo "Flipping LR_SYNTH_FLATTEN to 1"
+  export LR_SYNTH_FLATTEN=1
```

Comment:
I don't like this: If TIMING=1 and FLATTEN=0 is not allowed, I think it's probably clearer just to exit with an error message.

Reply:
Sounds okay to me. Did change the script accordingly.